convert2rhel/toolopts.py

```diff
@@ -345,7 +348,36 @@ def _process_cli_options(self):
                     "Ignoring the --serverurl option. It has no effect when --disable-submgr or --no-rhsm is used."
                 )
             else:
-                tool_opts.serverurl = parsed_opts.serverurl
+                # Parse the serverurl and save the components.
+                try:
+                    url_parts = _parse_subscription_manager_serverurl(parsed_opts.serverurl)
+                    url_parts = _validate_serverurl_parsing(url_parts)
+                except ValueError as e:
+                    # If we fail to parse, fail the conversion. The reason for
+                    # this harsh treatment is that we will be submitting
+                    # credentials to the server parsed from the serverurl. If
+                    # the user is specifying an internal subscription-manager
+                    # server but typo the url, we would fallback to the
+                    # public red hat subscription-manager server. That would
+                    # mean the user thinks the credentials are being passed
+                    # to their internal subscription-manager server but it
+                    # would really be passed externally.  That's not a good
+                    # security practice.
+                    loggerinst.critical(
+                        "Failed to parse a valid subscription-manager server from the --serverurl option.\n"
+                        "Please check for typos and run convert2rhel again with a corrected --serverurl.\n"
+                        "Supplied serverurl: %s\nError: %s" % (parsed_opts.serverurl, e)
+                    )
+
+                tool_opts.rhsm_hostname = url_parts.hostname
+
+                if url_parts.port:
+                    # urllib.parse.urlsplit() convert2 this into an int but we
+                    # always use it as a str
+                    tool_opts.rhsm_port = str(url_parts.port) if url_parts.port else ""
```

Comment:
```suggestion
                    tool_opts.rhsm_port = str(url_parts.port)
```
This line already is under the condition `if url_parts.port:`.

Reply:
Good point.  I have a vague recollection that the `else ""` might be needed so I changed the code to not do `if url_parts.port` first.